web/src/components/TestResults/AddTestSpecButton.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {useCallback, useMemo} from 'react';
+import {CaretDownOutlined} from '@ant-design/icons';
+import {Dropdown, Menu} from 'antd';
+import SpanService from 'services/Span.service';
+import Span from 'models/Span.model';
+import * as S from './TestResults.styled';
+import {useTestSpecForm} from '../TestSpecForm/TestSpecForm.provider';
+import {TEST_SPEC_SNIPPETS, TSnippet} from '../../constants/TestSpecs.constants';
+
+interface IProps {
+  selectedSpan: Span;
+}
+
+const AddTestSpecButton = ({selectedSpan}: IProps) => {
+  const {open} = useTestSpecForm();
+  const handleEmptyTestSpec = useCallback(() => {
+    const selector = SpanService.getSelectorInformation(selectedSpan);
+
+    open({
+      isEditing: false,
+      selector,
+      defaultValues: {
+        selector,
+      },
+    });
+  }, [open, selectedSpan]);
+
+  const onSnippetClick = useCallback(
+    (snippet: TSnippet) => {
+      open({
+        isEditing: false,
+        selector: snippet.selector,
+        defaultValues: snippet,
+      });
+    },
+    [open]
+  );
+
+  const menu = useMemo(
+    () => (
+      <Menu
+        items={[
+          {
+            label: 'Try these snippets for quick testing:',
+            key: 'test',
+            type: 'group',
+            children: TEST_SPEC_SNIPPETS.map(snippet => ({
+              label: snippet.name,
+              key: snippet.name,
+              onClick: () => onSnippetClick(snippet),
+            })),
```

Comment:
I was thinking if we should include an option to create an empty test spec. I know you can click the left part of the button but maybe users won't get it.

![Screenshot 2023-04-11 at 14 08 33](https://user-images.githubusercontent.com/3879892/231266249-47945eeb-2e51-4464-9bf2-bdf79c5d463f.png)

Reply:
Would it behave differently than clicking the left button? I think we can gather some feedback on the current UX and then we can think of updating it.

We also need @kdhamric feedback on what other snippets he'd like to add

Reply:
It should have the same behavior as clicking the left button. It could be the last item in the list (with a visual separator).

Reply:
Here are 3 more 'prompted test specs'

export const TRIGGER_SPAN_RESPONSE_BODY_CONTAINS: TSnippet = {
    name: 'Trigger Span: Response body contains "this string"',
    selector: 'span[tracetest.span.type="general" name="Tracetest trigger"]',
    assertions: [
      {
        left: 'attr:tracetest.response.body',
        comparator: 'contains',
        right: '"this string"',
      },
    ],
  };


  export const GRPC_SPANS_STATUS_CODE: TSnippet = {
    name: 'All gRPC Spans: Status is Ok',
    selector: 'span[tracetest.span.type="rpc" rpc.system="grpc"]',
    assertions: [
      {
        left: 'attr:grpc.status_code',
        comparator: '=',
        right: '0',
      },
    ],
  };


  export const DB_SPANS_QUALITY_DB_STATEMENT_PRESENT: TSnippet = {
    name: 'All Database Spans: db.statement should always be defined (QUALITY)',
    selector: 'span[tracetest.span.type="database"]',
    assertions: [
      {
        left: 'attr:db.system',
        comparator: '!=',
        right: '""',
      },
    ],
  };